apps/microsoft-teams/app-actions/src/actions/handle-app-event.ts

```diff
@@ -21,14 +25,63 @@ export const handler = withAsyncAppActionErrorHandling(
   async (
     parameters: AppActionCallParameters,
     context: AppActionCallContext
-  ): Promise<AppActionCallResponse<SendEntryActivityMessageResult[]>> => {
+  ): Promise<
+    AppActionCallResponse<SendEntryActivityMessageResult[] | SendWorkflowUpdateMessageResult>
+  > => {
     const {
       cma,
       appActionCallContext: { appInstallationId, environmentId, spaceId, userId, cmaHost },
     } = context;
 
     const { payload, topic: topicString, eventDatetime } = parameters;
 
+    /**
+     * In the interest of time, as well as prototyping/learning we made the explicit choice
+     * to re-use/extend this "app-events" app-action to handle "workflow" events as well.
+     *
+     * This block of code to handle workflow topics is intentionally repetative and intrusive.
+     * The end goal is to create a new app action to specifically handle "workflow" events.
+     * Ideally, at that time, it will be a simple copy/paste operation, leaving this
+     * app-action unchanged.
+     */
+    if (topicString === TOPIC_ACTION_MAP['Workflow.Step.notifyMicrosoftTeams']) {
+      const appInstallation = await cma.appInstallation.get({ appDefinitionId: appInstallationId });
+      const { tenantId } = parametersFromAppInstallation(appInstallation);
+
+      // request body
+      const entry = JSON.parse(payload) as EntryProps;
+
+      const entryActivity = await helpers.buildEntryActivity(
+        { entry, topic: topicString, eventDatetime },
+        cma,
+        cmaHost
+      );
+
+      const title = entryActivity.entryTitle;
+      const contentTypeId = entry.sys.contentType.sys.id;
+
+      const workflowUpdateMessage: WorkflowUpdateMessage = {
+        title: title,
+        contentType: contentTypeId,
+        currentStep: 'currentStep',
```

Comment:
Mabye drop a comment or TODO here like "TODO use actual values from workflow action"

Reply:
Sounds good

- [ ] "todo: ..."

Reply:
Actually, I'll investigate how this would actually be composed, i.e. what's the exact payload coming from `workflow-consumer-api`.  If it's already working, it might make sense to grab the actual data. 🤞